src/codegen/sys/lib_.rs

```diff
@@ -333,8 +333,11 @@ fn generate_unions(w: &mut Write, env: &Env, unions: &[&Union]) -> Result<()> {
         if !env.type_status_sys(&full_name).need_generate() {
             continue;
         }
+        let config = env.config.objects.get(&full_name);
+        let align = config.and_then(|c| c.align);
```

Comment:
Can you move `config` acquisition before status check and use it instead `env.type_status_sys`,
maybe with some new function in `Env`
```
pub fn unwrap_status_sys(status: Option<GStatus>) -> GStatus {
    status.unwrap_or(GStatus::Generate)
}
```

Reply:
.. or in global namespace with code near `GStatus` definition

Reply:
Sorry, forgot about `.map(|o| o.status)`
```
pub fn unwrap_status_sys(object: Option<GObject>) -> GStatus {
    if let Some(object) = object {
       object.status
    } else {
        GStatus::Generate
    }
}
```

Reply:
Other code in this file does it in a way that makes it easier to spot what's going on without increasing LOC:

```rust
if let Some(false) = config.map(|c| c.status.need_generate()) {
    continue;
}
```

I actually prefer this one. Let me know if you still want `unwrap_status_sys()`.

Reply:
This is better solution and `Env::type_status_sys` maybe unneeded now. Thanks.

Reply:
It was almost year ago :wink:
https://github.com/gtk-rs/gir/commit/032ec3fcc3cea62166c7a1cd941ff38946a18115#diff-0c37b0c87c3c91119e579feee853b5d0R187